Remotion/Core/Collections.Caching/CacheFactory.cs

```diff
@@ -276,7 +289,9 @@ public static class CacheFactory
     /// should therefor be replaced.
     /// </remarks>
     [Obsolete ("Use CreateWithSynchronization(...) instead. (Version: 1.19.3)")]
-    public static ICache<TKey, TValue> CreateWithLazyLocking<TKey, TValue> () where TValue : class
+    public static ICache<TKey, TValue> CreateWithLazyLocking<TKey, TValue> () 
+        where TValue : class
+        where TKey : notnull

```

Comment:
reorder constraints here, next line and in the datastore-PR, too. Preferred order is  always to follow the declaration order on the type.